### PR TITLE
Fix bug in static refinement

### PR DESCRIPTION
**Pull request by lroberts36**
## PR Summary

The re-definition of `Mesh::root_level` introduced in #1053 did not get correctly propagated through the static refinement related code (which operates on logical locations relative to the legacy tree). This PR fixes that issue, which had cropped up in Artemis.

## PR Checklist

- [x] #1073 is merged
- [x] Code passes cpplint
- [ ] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [x] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] Docs build
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by Yurlungur — COMMENTED** [review]
Thanks for the catch! To prevent this kind of thing from happening again... maybe we should have a regression test that uses SMR?

**Comment by lroberts36** [comment]
Agreed, we need to add a couple tests that use multi-tree forests. We have at least one that uses SMR, but the bug does not manifest if the base grid only requires one tree.

File: tst/regression/test_suites/sparse_advection/sparse_advection.py [hunk 1]
```diff
@@ -33,6 +33,27 @@ def Prepare(self, parameters, step):
...
+                "parthenon/mesh/nx2=32",
+                "parthenon/time/nlim=50",
+                "parthenon/job/problem_id=sparse_twotree_static",
+                "parthenon/mesh/refinement=static",
+                "parthenon/static_refinement0/x1min=-0.75",
+                "parthenon/static_refinement0/x1max=-0.5",
+                "parthenon/static_refinement0/x2min=-0.75",
+                "parthenon/static_refinement0/x2max=-0.5",
+                "parthenon/static_refinement0/level=3",
+            ]
```
Comment (Yurlungur) [inline tst/regression/test_suites/sparse_advection/sparse_advection.py:55]: :+1:

**Review by Yurlungur — APPROVED** [review]
:+1: for fixes and :+1: for tests

**Review by pdmullen — APPROVED** [review]
LGTM

File: tst/regression/test_suites/sparse_advection/sparse_advection.py [hunk 2]
```diff
@@ -77,5 +98,31 @@ def Analyse(self, parameters):
...
+                return False
+
+            delta = compare(
+                [
+                    "sparse_twotree.out0.final.phdf",
+                    parameters.parthenon_path
+                    + "/tst/regression/gold_standard/sparse_twotree.out0.final.phdf",
+                ],
+                one=True,
+                tol=1e-12,
```
Comment (pgrete) [inline tst/regression/test_suites/sparse_advection/sparse_advection.py:112]: Any reason for the "loose" tolerance?

File: tst/regression/test_suites/sparse_advection/sparse_advection.py [hunk 3]
```diff
@@ -77,5 +98,31 @@ def Analyse(self, parameters):
                 tol=1e-12,
                 check_metadata=False,
             )
+            if delta != 0:
+                return False
```
Comment (pgrete) [inline tst/regression/test_suites/sparse_advection/sparse_advection.py]: In other tests with num_steps > 0, this approach sometimes resulted in some difficulties in figuring out where the test actually failed.
So I suggest to either add a `print("check X failed")` before each `return` or add some `passed = False` with a `print` and only a single `return passed` at the end of the `Analyse` function.

**Review by pgrete — APPROVED** [review]
LGTM (plus optional comments on the python stuff)

The updated gold files are in #1073, right?

File: tst/regression/test_suites/sparse_advection/sparse_advection.py (hunk 2 above)
Comment (lroberts36) [reply tst/regression/test_suites/sparse_advection/sparse_advection.py:112]: I think I just copied this tolerance from the other checks. Isn't 1.e-12 what we set for most tests? I don't think this test should be bitwise exact between machines.

File: tst/regression/test_suites/sparse_advection/sparse_advection.py (hunk 3 above)
Comment (lroberts36) [reply tst/regression/test_suites/sparse_advection/sparse_advection.py]: added.

**Comment by lroberts36** [comment]
> The updated gold files are in #1073, right?

Yeah, I added the new gold files required here (and in #1047) to v23 so that I didn't have to upgold a bunch of times.

**Pull request merged**